src/components/page/WithGATracker/index.tsx

```diff
@@ -30,32 +38,49 @@ export const getGAusername = (): string => username;
  * @param {string} page the url string of the page view being tracked
  * @param {FlexObject} options tracking options for the page view
  */
-export const trackPage = (page: string, options: FlexObject = {}): void => {
-  GoogleAnalytics.set({
-    page,
-    ...options,
-  });
-  GoogleAnalytics.pageview(page);
+export const trackPage = (page: string, options: TrackingOptions): void => {
+  const { GA_CODE } = options;
+  if (GA_CODE && GA_CODE.length) {
+    GoogleAnalytics.set({
+      page,
+      ...options,
+    });
+    GoogleAnalytics.pageview(page);
+  }
 };
 
-if (GA_CODE.length) {
-  GoogleAnalytics.initialize(GA_CODE, {
-    testMode: GA_ENV === 'test',
-  });
-  GoogleAnalytics.set({
-    env: GA_ENV,
-    username,
-  });
-}
+/**
+ * helper function to initialize GoogleAnalytics
+ * @param {TrackingOptions} options tracking options for the page view
+ * @returns {TrackingOptions}
+ */
+export const initGoogleAnalytics = (options: TrackingOptions): TrackingOptions => {
+  const { GA_CODE, GA_ENV } = options;
+  if (GA_CODE && GA_CODE.length) {
+    GoogleAnalytics.initialize(GA_CODE, {
+      testMode: GA_ENV === TEST,
+    });
+    GoogleAnalytics.set({
+      env: GA_ENV || TEST,
+      username,
+    });
+  }
+  return options;
+};
 
 /**
  * Higher Order Component (HOC) which handles Google Analytics page view tracking
  * @param {any} WrappedComponent the component to be wrapped by the HOC component
- * @param {FlexObject} options tracking options for the page view
+ * @param {TrackingOptions} options tracking options for the page view
  * @returns HOC rendering the WrappedComponent
  */
-const WithGATracker = (WrappedComponent: any, options: FlexObject = {}) => {
-  const HOC = class extends Component<Props> {
+const WithGATracker = (WrappedComponent: any, options: TrackingOptions) => {
```

Comment:
@cKellyDesign how do you feel about giving options a default value here, to avoid the `component={WithGATracker(ConnectedPractitionersListView, trackingOptions)}` kind of changes in `App.tsx`?

Reply:
ah ok good idea, think those defaults should pull from the global env then?

Reply:
Lets also not use `any` here for `WrappedComponent` - I believe there is a type for React Components that we can use e.g. the `React.ElementType` kind of types from React

Reply:
> ah ok good idea, think those defaults should pull from the global env then?

Yeah

Reply:
think i got it :)